### PR TITLE
pmix: add explicit hwloc dependency

### PR DESCRIPTION
**Pull request by sethrj**
pmix@:2 uses hwloc if it's available (e.g. from homebrew). Explicitly enable it and use version @1 for pmix:@2.

### CODE REVIEW

**Comment by rhc54** [comment]
I'm afraid I don't know anything about Spack syntax, so I can't help there. I can say that more recent versions of PMIx (I believe starting in the v4.x series) _require_ HWLOC - it is no longer optional. Is that reflected here?

**Comment by sethrj** [comment]
@rhc54 Yes. It's required for new versions, and the problem is that since it was "optional" for old versions the configure script grabs a system hwloc that's too new. So to make things easy I'm requiring it for the old version and adding the older version constraint

**Review by rhc54 — APPROVED** [review]

**Comment by rhc54** [comment]
Makes sense - thx!

**Comment by sethrj** [comment]
@haampie this needs a maintainer approval; since you've worked with openmpi/pmix/hwloc do you mind?

**Review by haampie — APPROVED** [review]

**Pull request merged**
